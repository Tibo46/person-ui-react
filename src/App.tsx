import React from 'react';
import Routes from './Routes';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider, createMuiTheme, CssBaseline } from '@material-ui/core';

const theme = createMuiTheme({
  palette: {
    primary: { main: '#EF4F84' },
    secondary: { main: '#344955' },
    background: {
      default: '#EDF0F3',
    },
  },
  overrides: {
    MuiTypography: {
      h3: {
        marginTop: '2rem',
        marginBottom: '1.5rem',
        fontSize: '1rem',
        fontStyle: 'italic',
      },
    },
    MuiInputLabel: {
      shrink: {
        transform: 'translate(14px, -15px) scale(0.75)!important',
      },
    },
    MuiPopover: {
      paper: {
        borderRadius: '10px',
        background: '#edf0f3',
        boxShadow: '14px 18px 25px #bec0c2, -4px -8px 25px #ffffff',
      },
    },
    MuiSelect: {
      selectMenu: {
        borderRadius: '10px',
        minWidth: '50px',
        '&:focus': {
          borderRadius: '10px',
        },
      },
    },
    MuiOutlinedInput: {
      root: {
        '& fieldset': {
          border: 'none',
        },
      },
      multiline: {
        padding: '0',
        '& textarea': {
          padding: '18.5px 14px',
        },
      },
      input: {
        borderRadius: '10px',
        background: '#EDF0F3',
        color: '#3b3f44',
        fontWeight: 500,
        letterSpacing: '.5px',
        boxShadow: 'inset -1px 1px 5px #bec0c2, inset 6px -6px 19px #ffffff',
        '&:invalid': {
          background: '#EDF0F3',
          boxShadow: 'inset -1px 1px 5px #bec0c2, inset 6px -6px 19px #ffffff',
        },
        '&:active,&:focus': {
          background: '#EDF0F3',
          boxShadow: 'inset -6px 6px 5px #bec0c2, inset 6px -6px 19px #ffffff',
        },
      },
    },
    MuiButton: {
      containedPrimary: {
        color: '#fff',
        borderRadius: '35px',
        backgroundColor: '#EF4F84',
        boxShadow:
          '-7px -7px 20px 0px #fff9, -4px -4px 5px 0px #fff9, 7px 7px 20px 0px #0002, 4px 4px 5px 0px #0001, inset 0px 0px 0px 0px #fff9, inset 0px 0px 0px 0px #0001, inset 0px 0px 0px 0px #fff9, inset 0px 0px 0px 0px #0001',

        '&:hover': {
          backgroundColor: '#e42a68',
          boxShadow:
            '-7px -7px 20px 0px #fff9, -4px -4px 5px 0px #fff9, 7px 7px 20px 0px #0002, 4px 4px 5px 0px #0001, inset 0px 0px 0px 0px #fff9, inset 0px 0px 0px 0px #0001, inset 0px 0px 0px 0px #fff9, inset 0px 0px 0px 0px #0001',
        },
        '&:focus:active': {
          background: '#e42a68',
          boxShadow:
            '0px 0px 0px 0px #fff9, 0px 0px 0px 0px #fff9, 0px 0px 0px 0px #0001, 0px 0px 0px 0px #0001, inset -7px -7px 20px 0px #e42a68, inset -4px -4px 5px 0px #fff9, inset 7px 7px 9px 0px rgba(0, 0, 0, 0.5), inset 4px 4px 5px 0px #0001',
        },
        '&:focus-only': {
          background: '#e42a68',
          boxShadow:
            '0px 0px 0px 0px #fff9, 0px 0px 0px 0px #fff9, 0px 0px 0px 0px #0001, 0px 0px 0px 0px #0001, inset -7px -7px 20px 0px #e42a68, inset -4px -4px 5px 0px #fff9, inset 7px 7px 9px 0px rgba(0, 0, 0, 0.5), inset 4px 4px 5px 0px #0001',
        },
      },
      containedSecondary: {
        color: '#000',
        borderRadius: '35px',
        backgroundColor: '#EDF0F3',
        boxShadow:
          '-7px -7px 20px 0px #fff9, -4px -4px 5px 0px #fff9, 7px 7px 20px 0px #0002, 4px 4px 5px 0px #0001, inset 0px 0px 0px 0px #fff9, inset 0px 0px 0px 0px #0001, inset 0px 0px 0px 0px #fff9, inset 0px 0px 0px 0px #0001',

        '&:hover': {
          backgroundColor: '#e4e4e4',
          boxShadow:
            '-7px -7px 20px 0px #fff9, -4px -4px 5px 0px #fff9, 7px 7px 20px 0px #0002, 4px 4px 5px 0px #0001, inset 0px 0px 0px 0px #fff9, inset 0px 0px 0px 0px #0001, inset 0px 0px 0px 0px #fff9, inset 0px 0px 0px 0px #0001',
        },
        '&:focus:active': {
          background: '#e0e4e8',
          boxShadow:
            '0px 0px 0px 0px #fff9, 0px 0px 0px 0px #fff9, 0px 0px 0px 0px #0001, 0px 0px 0px 0px #0001, inset -7px -7px 20px 0px #fff9, inset -4px -4px 5px 0px #fff9, inset 7px 7px 20px 0px #0003, inset 4px 4px 5px 0px #0001',
        },
        '&:focus-only': {
          background: '#e0e4e8',
          boxShadow:
            '0px 0px 0px 0px #fff9, 0px 0px 0px 0px #fff9, 0px 0px 0px 0px #0001, 0px 0px 0px 0px #0001, inset -7px -7px 20px 0px #fff9, inset -4px -4px 5px 0px #fff9, inset 7px 7px 20px 0px #0003, inset 4px 4px 5px 0px #0001',
        },
      },
    },
  },
});

function App() {
  document.body.classList.remove('loading');

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <Router>
        <Routes />
      </Router>
    </ThemeProvider>
  );
}

export default App;
