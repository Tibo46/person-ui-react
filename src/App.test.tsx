import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('App', () => {
  it('App should render', () => {
    const component = renderer.create(<App />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
