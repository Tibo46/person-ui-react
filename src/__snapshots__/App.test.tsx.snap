// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should render 1`] = `
<div
  className="makeStyles-root"
>
  <header
    className="makeStyles-mobileHeader"
  >
    <div
      className="makeStyles-searchBtn"
    >
      <mui-icon-button
        onClick={[Function]}
      >
        <mui-icon-search />
      </mui-icon-button>
    </div>
  </header>
  <mui-drawer
    className="makeStyles-drawer"
    variant="permanent"
  >
    <div
      className="makeStyles-toolbar"
    >
      <a
        className="makeStyles-link makeStyles-homeLink"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Logo"
          className="makeStyles-logo"
          src="logo.png"
        />
      </a>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <ul
      className="MuiList-root MuiList-padding"
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root makeStyles-listButton MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <a
          className="makeStyles-link makeStyles-listItemLink"
          href="/employees"
          onClick={[Function]}
        >
          <mui-list-item-icon
            className="makeStyles-listItemIcon"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root makeStyles-listIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 10H7v2h10v-2zm2-7h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zm-5-5H7v2h7v-2z"
              />
            </svg>
          </mui-list-item-icon>
          <mui-list-item-text
            primary="All Employees"
          />
        </a>
      </div>
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root makeStyles-listButton MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <a
          className="makeStyles-link makeStyles-listItemLink"
          href="/employees/add"
          onClick={[Function]}
        >
          <mui-list-item-icon
            className="makeStyles-listItemIcon"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root makeStyles-listIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.53 11.06L15.47 10l-4.88 4.88-2.12-2.12-1.06 1.06L10.59 17l5.94-5.94zM19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11z"
              />
            </svg>
          </mui-list-item-icon>
          <mui-list-item-text
            primary="Add Employee"
          />
        </a>
      </div>
    </ul>
  </mui-drawer>
  <main
    className="makeStyles-container"
  >
    <div>
      <h1
        className="MuiTypography-root MuiTypography-h1"
      >
        Welcome to Employee Manager
      </h1>
      <h2
        className="MuiTypography-root MuiTypography-h2"
      >
        Manage your employees easily
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        You can use the search button on the top right corner to look for any specific employee or group of employees.
        <br />
        To see all of them, use the 
        <a
          className="makeStyles-link undefined"
          href="/employees"
          onClick={[Function]}
        >
          Employee page
        </a>
        .
        <br />
        If you have a new hire, don't forget to add him/her into the system via the
         
        <a
          className="makeStyles-link undefined"
          href="/employees/add"
          onClick={[Function]}
        >
          Employee Create page
        </a>
        !
      </p>
      <p
        className="MuiTypography-root makeStyles-covid MuiTypography-body2"
      >
        Don't forget to 
        <a
          className="makeStyles-link undefined"
          href="https://www.gov.uk/coronavirus"
        >
          stay home
        </a>
        !
      </p>
    </div>
  </main>
</div>
`;
